kbi/0022/index.rst

```diff
@@ -0,0 +1,103 @@
+.. index::
+   single: datalad save; git mv
+
+KBI0022: Performance of ``mv`` vs ``git mv`` when renaming dataset directories 
+==============================================================================
+
+:authors: Stephan Heunis <jsheunis@gmail.com>
+:discussion: <link>
+:keywords: performance, time, datalad save, mv, git move, rename
+:software-versions: datalad_0.18.4, git-annex_10.20230330-g98a3ba0ea
+
+When renaming a directory that forms part of many files in a large DataLad dataset,
+a subsequent ``datalad save`` may take an unexpected amount of time. While performance
+is always relative, it is worth considering the use of ``git mv`` followed by
+``git commit`` instead of a standard ``mv`` and ``datalad save`` in datasets with large
+trees.
+
+.. note::
+
+   The use of ``mv`` for renaming and moving dataset content is covered extensively
+   in a dedicated DataLad Handbook chapter: `Miscellaneous file system operations`_.
+   This also includes comparisons with ``git mv`` and comments on when and when not to
+   use either of these methods. Performance, however, is not covered in the handbook
+   and is hence presented here.
```

Comment:
```suggestion
   and is hence presented here.
   **Important:** The recommendations here solely apply to directories. ``git mv`` operations should `not be performed on subdatasets <https://github.com/datalad/datalad/issues/3464>`_. Instead, stick to a plain ``mv`` followed by a ``datalad save``. 
```

Reply:
I think its important to mention this here again. I took the opportunity to test if this long-known problem (https://github.com/datalad/datalad/issues/3464) persists, and it indeed does.